### PR TITLE
OBPIH-2509 Dashboard: Sent stock movements

### PR DESCRIPTION
**Pull request by bamoure**

### CODE REVIEW

File: grails-app/services/org/pih/warehouse/tableroapi/IndicatorDataService.groovy [hunk 1]
```diff
@@ -110,46 +110,40 @@ class IndicatorDataService {
...
-        List listData = []
-        listLabel = []
-        try {
-            List locationName = dataService.executeQuery("select name from location where id=" + item[0].value)
-            // querySize is the quantity of months in the filter : until which month query data
+    DataGraph getSentStockMovements(def location, def params) {
+        Integer querySize = params.querySize? params.querySize.toInteger() : 5
+        today.clearTime()
+
+        String query = "SELECT COUNT(s.id) as total, YEAR(s.last_updated) as year, s.destination_id as destination, MONTH(s.last_updated) as month FROM openboxes.shipment s WHERE origin_id = '" + location.id + "' AND s.current_status <> 'PENDING' GROUP BY MONTH(s.last_updated), s.destination_id, YEAR(s.last_updated)"
```
Comment (jmiranda) [inline grails-app/services/org/pih/warehouse/tableroapi/IndicatorDataService.groovy]: Remove the schema from the query (openboxes.shipment).

File: grails-app/services/org/pih/warehouse/tableroapi/IndicatorDataService.groovy [hunk 2]
```diff
@@ -110,46 +110,40 @@ class IndicatorDataService {
...
-                monthEnd.set(month: today.month - i + 1, date: 1)
+                Date month = today.clone()
+                month.set(month: today.month - i, date: 1)
 
-                def query = Shipment.executeQuery("select count(*) from Shipment s where s.lastUpdated >= :monthOne and s.lastUpdated < :monthTwo and s.origin = " + item[0].value + " and s.currentStatus <> 'PENDING'", 
-                // The column transaction_transaction_date doesn't exist, using lastUpdated instead
-                ['monthOne': monthBegin, 'monthTwo': monthEnd]);
-                String monthLabel = new java.text.DateFormatSymbols().months[monthBegin.month].substring(0,3)
+                Integer value = 0
+                if (month.month == item.month-1 && month.year + 1900 == item.year) {
```
Comment (jmiranda) [inline grails-app/services/org/pih/warehouse/tableroapi/IndicatorDataService.groovy]: Please add a comment. Depending on what you're trying to do I might suggest another approach.

Comment (jmiranda) [reply grails-app/services/org/pih/warehouse/tableroapi/IndicatorDataService.groovy]: In fact, can you comment on all of this ... the two loops as well.

File: src/js/components/tablero/GraphCard.jsx [hunk 3]
```diff
@@ -67,9 +67,9 @@ const GraphCard = SortableElement(({
             onChange={e => reloadIndicator(cardMethod, cardType, cardTitle, cardLink, cardId, `querySize=${e.target.value}`)}
             disabled={!filter}
           >
-            <option value="6">Last 6 Months</option>
-            <option value="12">Last Year</option>
-            <option value="24">Last 2 Years</option>
+            <option value="5">Last 6 Months</option>
```
Comment (jmiranda) [inline src/js/components/tablero/GraphCard.jsx]: You should leave it at the value expected and normalize where you need to.

**Comment by bamoure** [comment]
@jmiranda we made some changes but kept the loops to get the data in the right format for the graph

**Comment by jmiranda** [comment]
@bamoure Thank you the comments look great. I am going to merge for now and give you guys some suggestions on how to make the code a little more readable using some of the more useful Groovy collection constructs.

**Pull request merged**